candid/tests/parse_type.rs

```diff
@@ -12,7 +12,7 @@ fn parse_idl_prog() {
     let prog = r#"
 import "test.did";
 type my_type = nat8;
-type List = record { head: int; tail: List };
+type List = variant { head: int; tail: List };
```

Comment:
I guess this should be
```suggestion
type List = opt record { head: int; tail: List };
```

Reply:
It is very interesting, how many forms can one simple data structure have.
I vote for this.